web_widget_image_webcam/static/src/js/webcam_widget.js

```diff
@@ -0,0 +1,141 @@
+/* global Webcam */
+/*
+    Copyright 2016 Siddharth Bhalgami <siddharth.bhalgami@gmail.com>
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+*/
+odoo.define("web_widget_image_webcam.webcam_widget", function (require) {
+    "use strict";
+
+    var core = require("web.core");
+    var rpc = require("web.rpc");
+    var Dialog = require("web.Dialog");
+    const {patch} = require("@web/core/utils/patch");
+    const {ImageField} = require("@web/views/fields/image/image_field");
+
+    var _t = core._t;
+    var QWeb = core.qweb;
+
+    const getWebcamFlashFallbackModeConfig = rpc.query({
+        model: "ir.config_parameter",
+        method: "get_webcam_flash_fallback_mode_config",
+    });
+
+    patch(ImageField.prototype, "web_widget_image_webcam", {
+        setup() {
+            this._super.apply(this, arguments);
+            this.flash_fallback_mode = this._initWebcamFlashFallbackModeConfig();
```

Comment:
```suggestion
            this.flash_fallback_mode = this._initWebcamFlashFallbackModeConfig();
```

Since you're calling an async function, you'd have to await it in order to get the actual value.
As it is right now, `this.flash_fallback_mode` will be assigned a `Promise`.

You can only await within `async` functions, so it can't be done during `setup`.
Maybe the right place would be to override the `willStart` hook

```js
async willStart() {
    this.flash_fallback_mode = await this._initWebcamFlashFallbackModeConfig();
}
```.

or just do this and you could remove the `_initWebcamFlashFallbackModeConfig` method

```js
async willStart() {
    this.flash_fallback_mode = await getWebcamFlashFallbackModeConfig;
}
```


Alternatively, you could do everything on `_setWebcamParams` if you make that method `async`, because it seems that's the only place where it's used

Reply:
thanks for the explanation